comit/src/btsieve/ethereum/cache.rs

```diff
@@ -112,3 +114,29 @@ where
         Ok(receipt)
     }
 }
+
+/// We define, for the Web3 connector, a timestamp to have passed if the
+/// connector has seen a block with a block-timestamp smaller than the
+/// timestamp.
+#[async_trait]
+impl<C> HasPassed for Cache<C>
+where
+    C: LatestBlock<Block = Block>,
+{
+    async fn has_passed(&self, timestamp: Timestamp) -> bool {
+        match self.latest_block().await {
+            Err(e) => {
+                tracing::warn!("failed to get block time: {}", e);
+                false
```

Comment:
This error handling feels quite premature to me, I'd let the caller deal with this.

Reply:
Cool, we can return `Result<bool>` instead. @D4nte are you happy with that?

Reply:
Implemented as suggested.